aten/src/ATen/core/jit_type.h

```diff
@@ -2133,6 +2115,27 @@ static LayoutTypePtr get();
 LayoutType() : EnumerationType() {}
 };
 
+namespace detail {
```

Comment:
yeah, SymInt's specialization is here

```
template <>
struct getTypePtr_<SymInt> final {
  static decltype(auto) call() {
    return SymIntType::get();
  }
};
```

the ordering difference is because layout and co are enumeration types